sdk/core/Azure.Core/tests/HttpPipelineBuilderTest.cs

```diff
@@ -185,6 +190,32 @@ public async Task CustomClientRequestIdSetInPerCallPolicyAppliedAsAHeader()
             Assert.AreEqual("MyPolicyClientId", value);
         }
 
+        [Test]
+        public void LogWhenTransportOptionsProvidedWhenCustomTransportIsSet()
+        {
+            var transport = new MockTransport(new MockResponse(503), new MockResponse(200));
+            var options = new TestOptions
+            {
+                Transport = transport
+            };
+
+            List<EventWrittenEventArgs> events = new();
+
+            using var listener = new AzureEventSourceListener((args, s) =>
```

Comment:
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/tests/RedirectPolicyTests.cs#L25-L26